### ISSUE TITLE
Completing parameter values from a list of strings

### ISSUE THREAD

**Issue by timhoffm** [davidhalter/jedi#1355]
Could jedi be made to complete string parameters inferred from numpydoc docstrings?

~~~
def f(a): 
    """ 
    A function. 

    Parameters 
    ---------- 
    a : {'foo', 'bar'} 
    """ 
    print(a) 
~~~

When I complete `f(a=` or `f(a='` it would be really cool, if jedi could suggest 'foo' and 'bar'.

Would it be worth looking into this for a potential PR or are there fundamental road blocks?

**Comment by davidhalter** [comment]
I would be kind of fine with this once #771 was in place. Before this it won't really work, because the infrastructure is not in place.

**Comment by timhoffm** [comment]
Thanks for the feedback!

**Comment by davidhalter** [comment]
I fixed #771 and while this would be possible now, I'm very much against adding more complexity to the docstring parsers. It's just not really worth it for the little gain. There's a lot of issues with some of the numpy APIs IMO, so they should probably just use stubs.

I'm happy to reconsider this, if there are stubs out there for numpy (or related libraries) that use a `Union[Literal['foo'], Literal['bar']]`. See also: https://www.python.org/dev/peps/pep-0586/.

Annotations are the future, so using them makes only sense.
